src/AasxServerStandardBib/Services-DB/AssetAdministrationShellServiceDB.cs

```diff
@@ -0,0 +1,561 @@
+            }
+
+            var output = _packageEnvService.CreateAssetAdministrationShell(body);
+
+            return output;
+        }
+
+        public IReference CreateSubmodelReferenceInAAS(IReference body, string aasIdentifier)
+        {
+            IReference output = null;
+            //Verify request body
+            _verificationService.VerifyRequestBody(body);
+
+            // TODO (jtikekar, 2023-09-04): to check if submodel with requested submodelReference exists in the server
+            var aas = _packageEnvService.GetAssetAdministrationShellById(aasIdentifier, out _);
+
+            if (aas != null)
+            {
+                if (aas.Submodels.IsNullOrEmpty())
+                {
+                    aas.Submodels = new List<IReference>
+                    {
+                        body
+                    };
+                    output = aas.Submodels.Last();
+                }
+                else
+                {
+                    bool found = false;
+                    //Check if duplicate
+                    foreach (var submodelReference in aas.Submodels)
+                    {
+                        if (submodelReference.Matches(body))
+                        {
+                            found = true;
+                            break;
+                        }
+                    }
+
+                    if (found)
+                    {
+                        _logger.LogDebug($"Cannot create requested Submodel-Reference in the AAS !!");
+                        throw new DuplicateException($"Requested SubmodelReference already exists in the AAS with Id {aasIdentifier}.");
+                    }
+                    else
+                    {
+                        aas.Submodels.Add(body);
+                        output = aas.Submodels.Last();
+                    }
+                }
+            }
+
+            return output;
+        }
+
+
+
+        public void DeleteAssetAdministrationShellById(string aasIdentifier)
+        {
+            var aas = _packageEnvService.GetAssetAdministrationShellById(aasIdentifier, out int packageIndex);
+
+            if (aas != null && packageIndex != -1)
+            {
+                _packageEnvService.DeleteAssetAdministrationShell(packageIndex, aas);
+            }
+        }
+
+        public void DeleteFileByPath(string aasIdentifier, string submodelIdentifier, string idShortPath)
+        {
+            var found = IsSubmodelPresentWithinAAS(aasIdentifier, submodelIdentifier);
+            if (found)
+            {
+                _logger.LogDebug($"Found submodel with id {submodelIdentifier} in AAS with id {aasIdentifier}");
+                _submodelService.DeleteFileByPath(submodelIdentifier, idShortPath);
+            }
+            else
+            {
+                throw new($"Submodel with id {submodelIdentifier} NOT found in AAS with id {aasIdentifier}");
```

Comment:
Avoid using vague exception messages. Instead of throwing a generic exception, specify the type of exception being thrown for clarity and precision:

```csharp
throw new NotFoundException($"Submodel with id {submodelIdentifier} not found in AAS with id {aasIdentifier}");